src/pages/Swap/index.tsx

```diff
@@ -297,6 +311,114 @@ export default function Swap() {
 
   const [showAddRecipient, setShowAddRecipient] = useState<boolean>(false)
 
+  const SwapBoxButton = () => {
+    // Eco Router is computing the best route for the user, so we need to show a loading indicator
+    if (swapInfoIsLoading) {
+      return (
+        <ButtonPrimary style={{ textAlign: 'center' }} disabled>
+          Loading
+        </ButtonPrimary>
+      )
+    }
+
+    // Wallet is not connected, show connect button
+    if (!account) {
+      return <ButtonConnect />
+    }
+
+    // Wallet is connected
+    // User is trying to un/wrap
+
+    if (showWrap) {
+      return (
+        <ButtonPrimary disabled={Boolean(wrapInputError)} onClick={onWrap} data-testid="wrap-button">
+          {wrapInputError ?? (wrapType === WrapType.WRAP ? 'Wrap' : wrapType === WrapType.UNWRAP ? 'Unwrap' : null)}
+        </ButtonPrimary>
+      )
+    }
+
+    if (noRoute && userHasSpecifiedInputOutput) {
+      return (
+        <ButtonPrimary style={{ textAlign: 'center' }} disabled>
+          Insufficient liquidity
+        </ButtonPrimary>
+      )
+    }
+
+    if (showApproveFlow) {
+      return (
+        <RowBetween>
+          <ButtonConfirmed
+            onClick={approveCallback}
+            disabled={approval !== ApprovalState.NOT_APPROVED || approvalSubmitted}
+            width="48%"
```

Comment:
Just a reminder... We wanted to let the Buttons address these properties themselves instead of hardcoding them.
Non-blocking.